client/src/store/actions/authActions.tsx

```diff
@@ -31,3 +39,28 @@ export const loginUser = (
 };
 
 export const authStartLoading = () => ({ type: TYPES.LOGIN_LOADING });
+
+export const registerUser = (
+  registerData: RegisterData
+): ThunkAction<void, {}, {}, Action> => async dispatch => {
+  dispatch({
+    type: TYPES.REGISTER_LOADING
+  });
+
+  axios
+    .post("/api/users/register", registerData)
+    .then(resp => {
+      dispatch({
+        type: TYPES.REGISTER_SUCCESS,
+        token: resp.data.token
+      });
+    })
+    .catch(err => {
+      dispatch({
+        type: TYPES.REGISTER_ERROR,
+        errors: err.response.data.errors
+      });
+    });
+};
+
+export const registerStartLoading = () => ({ type: TYPES.REGISTER_LOADING });
```

Comment:
this line is unnecessary, same for authStartLoading, becouse we dispatching type directly inside login/register function

Reply:
Removed